### PR TITLE
sf2 redesign - attempt 2

### LINKED ISSUE #6454: Bad selection of colors on SF2 UI
### Bug Summary
Visual impaired can see the settings for BANK & PATCH.
Dark-purple on black is not functional

#### Steps to reproduce
Just look at it

#### Expected behavior
Clear and distinctable  UI-text

#### Actual behavior
Not distinctable  UI-text

#### Screenshot
image

#### Affected LMMS versions
< 1.2. 

#### Type of Issue
Good fist Issue

### PR DESCRIPTION
**Pull request by Rossmaxx**
Fixes #6454 and replaces #6473. 

This is how it looks (outdated). Suggestions welcome 

image

### CODE REVIEW

**Comment by YohananDiamond** [comment]
Some small critiques:
image
Other than that, it looks good! I just now wonder about the background - I personally like it but should it also be updated to follow the general design style that LMMS seems to be going towards?
By "this design" I mean stuff like the design in the Equalizer plugin:
image

**Comment by Rossmaxx** [comment]
> some small critiques: 

The image

I felt like doing that but was not sure. I'll give it a shot.

> should it also be updated to follow the general design style that LMMS seems to be going towards?

I tried but the background i did in the old pr was not good so just reverted it. Anyway I'll try again once i get more ideas.

**Comment by Monospace-V** [comment]
I'm actually quite a fan of the pinks in SF2. I feel like where orange-white:3o and warm brown: mallets, similarly phenolphthalein pinks: SF2 player.
As far as I understand, the only problems are the fact that bank and patch number is unreadable:
image

and maybe how the pink is too bright in the folder-browse icon:
image

I think this can be easily remedied by making a new folder icon if needed, and changing bank/patch selection.

Looking at all other instruments,
TripleOscillator: Orange-white, white dials on main instrument plugin settings.
sfxr: black and multicoloured, multicoloured dials.
organic: grey and green, dials are green (and violet/dark pink for harm to differentiate)
OpulenZ- peacock colour, whatever it's called; dials are white
SID- pale yellow, dials are some white

While I feel there's nothing wrong with the fact SF2 player is themed pink, as it can be a rather good looking colour if used well,
image
image
(I can't fix the icon, but presumably someone can.)

If you're going to make some parts of it green, then make all of it greenish, including the SF2 icon on the plugin sidebar and the text on top of the plugin. Having it green (whether to fit in with everything else or make it unique as an instrument) doesn't look good alongside pinks, and could be slightly visually unappealing.

**Comment by Rossmaxx** [comment]
>  It appears this is now an entire redesign (for reasons I don't entirely understand)

i decided to redesign the entire thing because changing the lcds to green feels off. i can't change the lcds to pink either cause the pink one is broken. 

>  The art appears a lot less comfortable within its style than old

fixed along with the other design flaws

**Comment by Monospace-V** [comment]
Then can I implore you to pick a different pink for LCDs instead of turning the entire plugin UI green?

The SF2 player design is similar to the design of some other LMMS plugins in terms of the partially 3d grey analogue knobby style.
Notice how the shadows and lights fall, the apparent depth.
Your design looks like only the GIG player.
image
_SF2 player. Notice how the interface looks solid, and how there are ridges and raised surfaces. This will be our reference._

image
_AudioFileProcessor, look at the box which contains the sample waveform- to be specific, its lower and right edges._

image
_Kicker. The interface speaks for itself._

image
_Mallets. Visibly 3d around the edges of the interface box within the interface._

image
_Monstro. The split between the tabs on the top and the instrument controls is a little jarring, being 2d and 3d respectively._

image
_Organic. Again, distinctly 3d analogue look to it, with the screws and lighting._

image
_GIG Player. 2d, plain because nobody else has looked at it in decades (who uses GIG files anymore?), inconsistent with the theming of every instrument having an at least unique if not 3d UI, but passable- again only because nobody uses GIG files._

image
_VeSTige. Look at the box which contains the name of the VST._

(It should be noted that I'm probably biased towards the current appearance of the plugins as they are, though. I'm wont to complain about any new designs, especially ones that look so fundamentally different from the previous UI, the UI of other plugins, and my idealized good appearance of a UI. Maybe this could look better in app? I'll download and check.)

What application do you use for the UI design, OP?

**Comment by Rossmaxx** [comment]
> What application do you use for the UI design, OP?

good ol' gimp.

> The SF2 player design is similar to the design of some other LMMS plugins in terms of the partially 3d grey analogue knobby style.

I agree but the for the new ui, I am looking at lost robot's compressor for inspiration. 

> Then can I implore you to pick a different pink for LCDs instead of turning the entire plugin UI green?

I asked the dev channels about the possibility before I started the work. The colour is not plain hex codes, instead the lcd num boxes use something called pixmaps which are not modifiable easily.

> It should be noted that I'm probably biased towards the current appearance of the plugins as they are, though. I'm wont to complain about any new designs, especially ones that look so fundamentally different from the previous UI, the UI of other plugins, and my idealized good appearance of a UI.

well, we all have our own tastes.

**Comment by Rossmaxx** [comment]
@YohananDiamond what'd you think of the new design?

**Comment by YohananDiamond** [comment]
Overall it looks nice, but here are some more nitpicks:
image
Also, is it possible to use the modern-style knobs? I think they would look nice here and they're already being used on LB302 & BitInvader, for example.
image
image

**Comment by Rossmaxx** [comment]
> Also, is it possible to use the modern-style knobs? 

I asked the devs in the server. Waiting for a response. 

> The background colour looks weird

I noticed that one but not really sure on what to replace it with. Your suggestion looks good. Will try to do that one.

> Sf2 looks like sfz

That's accidental but it looks good so i think it's fine.

> Black background on the file button

I don't think it will create any issues so no need to change that one. Will change if any problems come up 

> Blurry text.

It's because of gimp pixelating the text on low resolutions. Haven't found any workarounds yet  

Thanks yohana diamond for your suggestions

**Comment by YohananDiamond** [comment]
No problem!

> That's accidental but it looks good so i think it's fine.

I pointed it out because there's another format similar to sf2 called sfz, and it's not supported by LMMS (at least directly), so I think it would cause some confusion.

> I don't think it will create any issues so no need to change that one. Will change if any problems come up

I meant that the button right below has a gray background while this one has a black, and I found it kinda confusing

**Comment by Rossmaxx** [comment]
> Also, is it possible to use the modern-style knobs? I think they would look nice here and they're already being used on LB302 & BitInvader, for example.

done

**Comment by YohananDiamond** [comment]
Looks fine now! Though the open button background issue keeps on and maybe the knob could be slightly smaller (see comparison in the image)
image
Also, what if we made a background to put on where it's currently green? Something also green-ish but with some level of detail to make the plugin look more unique (which is a thing that I think it lost after redesigning it)
I'm gonna think about it and maybe I can come up with something soon, then I'll send it here

**Comment by RiedleroD** [comment]
I haven't got time to contribute rn, but imo a PCB design could fit the green well.

**Comment by Rossmaxx** [comment]
> I haven't got time to contribute rn, but imo a PCB design could fit the green well.

Noted. Will try when i get time

**Comment by Rossmaxx** [comment]
@YohananDiamond @RiedleroD looks good? any more suggestions.

**Comment by RiedleroD** [comment]
just a few tiny things. could you send me the gimp project file for that? I'll try to improve upon your design if that's ok.

**Comment by Rossmaxx** [comment]
do you have discord? please dm me rossmaxx#9915. also, the gimp project file is a bit of a mess.

**Comment by YohananDiamond** [comment]
Just checked it out and I think the idea looks good but it still needs some polishing.
- Again on the knobs: I noticed they aren't aligned with the text (I think this just involves offseting them horizontally a little bit more).
- The background is nice but I think some more lines would make things more vivid. These new lines could be a little bit closer to the background color so they don't catch the user's attention too much.
- Aliasing: too much on the button texts, too little on the big texts & the circles. I don't really know how to fix this properly, but here are some suggestions:
    - Make the yellow circles thicker and _maybe_ enable antialiasing;
    - Make it so the button texts are not bold;
    - Enable antialiasing on the "SF2" text;
but I think you can put antialiasing without fear on the circles and, for the fonts, also put antialiasing but, on the smaller ones

File: plugins/Sf2Player/Sf2Player.cpp [hunk 1]
```diff
@@ -1020,19 +1005,19 @@ Sf2InstrumentView::Sf2InstrumentView( Instrument * _instrument, QWidget * _paren
 	m_chorusButton->setInactiveGraphic( PLUGIN_NAME::getIconPixmap( "chorus_off" ) );
 	m_chorusButton->setToolTip(tr("Apply chorus (if supported)"));
 
-	m_chorusNumKnob = new Sf2Knob( this );
+	m_chorusNumKnob = new Knob( knobBright_26, this );
```
Comment (sakertooth) [inline plugins/Sf2Player/Sf2Player.cpp:1008]:
```suggestion
	m_chorusNumKnob = new Knob(knobBright_26, this);
```

File: plugins/Sf2Player/Sf2Player.cpp [hunk 2]
```diff
@@ -1020,19 +1005,19 @@ Sf2InstrumentView::Sf2InstrumentView( Instrument * _instrument, QWidget * _paren
...
 	m_chorusNumKnob->setHintText( tr("Voices:"), "" );
 	m_chorusNumKnob->move( 93, 206 );
 
-	m_chorusLevelKnob = new Sf2Knob( this );
+	m_chorusLevelKnob = new Knob( knobBright_26, this );
 	m_chorusLevelKnob->setHintText( tr("Level:"), "" );
 	m_chorusLevelKnob->move( 130 , 206 );
 
-	m_chorusSpeedKnob = new Sf2Knob( this );
+	m_chorusSpeedKnob = new Knob( knobBright_26, this );
```
Comment (sakertooth) [inline plugins/Sf2Player/Sf2Player.cpp:1016]:
```suggestion
	m_chorusSpeedKnob = new Knob(knobBright_26, this);
```

File: plugins/Sf2Player/Sf2Player.cpp [hunk 3]
```diff
@@ -1020,19 +1005,19 @@ Sf2InstrumentView::Sf2InstrumentView( Instrument * _instrument, QWidget * _paren
...
 	m_chorusSpeedKnob->setHintText( tr("Speed:"), "" );
 	m_chorusSpeedKnob->move( 167 , 206 );
 
-	m_chorusDepthKnob = new Sf2Knob( this );
+	m_chorusDepthKnob = new Knob( knobBright_26, this );
```
Comment (sakertooth) [inline plugins/Sf2Player/Sf2Player.cpp:1020]:
```suggestion
	m_chorusDepthKnob = new Knob(knobBright_26, this);
```

**Review by sakertooth — CHANGES_REQUESTED** [review]
Just some formatting adjustments.

File: plugins/Sf2Player/Sf2Player.cpp [hunk 4]
```diff
@@ -938,12 +923,12 @@ Sf2InstrumentView::Sf2InstrumentView( Instrument * _instrument, QWidget * _paren
 
 
 	// LCDs
-	m_bankNumLcd = new LcdSpinBox( 3, "21pink", this );
+	m_bankNumLcd = new LcdSpinBox( 3, "19green", this );
```
[the remark this answers is not in the document]
Comment (Rossmaxx) [reply plugins/Sf2Player/Sf2Player.cpp]: Fixed all the formatting issues.

**Review by sakertooth — COMMENTED** [review]
Just to resolve the accidental request for review from @Rossmaxx (He informed me that this was the case).

**Comment by Rossmaxx** [comment]
The blurry text has been fixed. I installed and used inkscape for that

**Comment by Rossmaxx** [comment]
closing this cause i don't got time to get it finished.

**Pull request closed without merging**